pkgs/by-name/qb/qbittorrent-enhanced/package.nix

```
@@ -1,48 +1,95 @@
+  nix-update-script,
+  openssl,
+  pkg-config,
+  python3,
+  qt6,
+  stdenv,
+  trackerSearch ? true,
+  webuiSupport ? true,
+  wrapGAppsHook3,
+  zlib,
 }:
 
-stdenv.mkDerivation rec {
-  pname = "qbittorrent-enhanced";
-  version = "4.6.7.10";
+stdenv.mkDerivation (finalAttrs: {
+  pname = "qbittorrent-enhanced" + lib.optionalString (!guiSupport) "-nox";
+  version = "5.0.2.10";
 
   src = fetchFromGitHub {
     owner = "c0re100";
     repo = "qBittorrent-Enhanced-Edition";
-    rev = "release-${version}";
-    hash = "sha256-Mr376nH2pE6NMF9IdDhZULwqkRr7472OZnKO86hb6ZI=";
+    rev = "release-${finalAttrs.version}";
+    hash = "sha256-9RCG530zWQ+qzP0Y+y69NFlBWVA8GT29dY8aC1cvq7o=";
   };
 
   nativeBuildInputs = [
-    pkg-config
     cmake
-    qt5.wrapQtAppsHook
+    pkg-config
+    wrapGAppsHook3
+    qt6.wrapQtAppsHook
   ];
 
-  buildInputs = [
-    openssl.dev
-    boost
-    zlib
-    libtorrent-rasterbar
-    qt5.qtbase
-    qt5.qttools
-  ];
+  buildInputs =
+    [
+      boost
+      hicolor-icon-theme
+      libtorrent-rasterbar
+      openssl
+      qt6.qtbase
+      qt6.qtsvg
+      qt6.qttools
+      zlib
+    ]
+    ++ lib.optionals stdenv.hostPlatform.isDarwin [ apple-sdk_11 ]
+    ++ lib.optionals guiSupport [ dbus ]
+    ++ lib.optionals (guiSupport && stdenv.hostPlatform.isLinux) [ qt6.qtwayland ]
+    ++ lib.optionals trackerSearch [ python3 ];
+
+  cmakeFlags =
+    [ "-DVERBOSE_CONFIGURE=ON" ]
+    ++ lib.optionals (!guiSupport) [
+      "-DGUI=OFF"
+      "-DSYSTEMD=ON"
+      "-DSYSTEMD_SERVICES_INSTALL_DIR=${placeholder "out"}/lib/systemd/system"
+    ]
+    ++ lib.optionals (!webuiSupport) [ "-DWEBUI=OFF" ];
+
+  qtWrapperArgs = lib.optionals trackerSearch [ "--prefix PATH : ${lib.makeBinPath [ python3 ]}" ];
+
+  dontWrapGApps = true;
+
+  postInstall = lib.optionalString stdenv.hostPlatform.isDarwin ''
+    APP_NAME=qbittorrent-enhanced${lib.optionalString (!guiSupport) "-nox"}
+    mkdir -p $out/{Applications,bin}
+    mv $out/$APP_NAME.app $out/Applications
+    makeWrapper $out/{Applications/$APP_NAME.app/Contents/MacOS,bin}/$APP_NAME
+  '';
+
+  preFixup = ''
+    qtWrapperArgs+=("''${gappsWrapperArgs[@]}")
+  '';
+
+  passthru.updateScript = nix-update-script { };
 
   meta = {
     description = "Unofficial enhanced version of qBittorrent, a BitTorrent client";
     homepage = "https://github.com/c0re100/qBittorrent-Enhanced-Edition";
-    changelog = "https://github.com/c0re100/qBittorrent-Enhanced-Edition/blob/${src.rev}/Changelog";
-    license = with lib.licenses; [ gpl2Only gpl3Only ];
+    changelog = "https://github.com/c0re100/qBittorrent-Enhanced-Edition/blob/release-${finalAttrs.version}/Changelog";
+    license = with lib.licenses; [
+      gpl2Only
+      gpl3Only
+    ];
     maintainers = with lib.maintainers; [ ByteSudoer ];
-    mainProgram = "qBittorrent-enhanced";
-    platforms = lib.platforms.linux;
+    mainProgram = "qbittorrent-enhanced" + lib.optionalString (!guiSupport) "-nox";
```

```suggestion
    mainProgram = "qbittorrent" + lib.optionalString (!guiSupport) "-nox";
```

Since qbittorrent-enhanced is a drop-in replacement, I think it's fine to keep the binary the same as qbittorrent.

This will prevent nix-env to install both qbittorrent and qbittorrent-enhanced (can not link same binary in $PATH), for good reason.

I assume qbittorrent-enhanced will use the same data and config dir as qbittorrent, if user run both of them at the same time, it may cause problems.